search-attributes/src/workflows.ts

```diff
@@ -1,3 +1,14 @@
+import { upsertSearchAttributes, workflowInfo } from '@temporalio/workflow';
+
+// @@@SNIPSTART typescript-search-attributes-workflow
 export async function example(): Promise<string> {
-  return 'hello workflow';
+  const customInt = (workflowInfo().searchAttributes?.CustomIntField[0] as number) || 0;
```

Comment:
Aren't search attributes always defined on workflowInfo? They should be

Reply:
Shall we set to empty object on start? 
<img width="1482" alt="image" src="https://user-images.githubusercontent.com/251288/176566720-8402ed49-ed95-43dd-8a49-03d59a44c9a2.png">

Reply:
Yes

Reply:
Looking at the sample code, I'd make the SearchAttributes type `Record<string, SearchAttributeValue | undefined>` so TS doesn't assume that any key will be present in the map

Reply:
Yeah. https://github.com/temporalio/sdk-typescript/pull/738